### PR TITLE
Split sdks and nm into separate packages from @yarnpkg-pnpify

### PR DESCRIPTION
**Pull request by larixer**
**What's the problem this PR addresses?**

The `@yarnpkg/pnpify` package contains 3 different codebases:
1. Editor integrations for PnP
2. node_modules tree hoister
3. Pnpify tool that creates in-memory node_modules to run non-PnP enabled JS tools in PnP projects

**How did you fix it?**

I have moved node-modules hoisting and tree building code into separate `@yarnpkg/nm` package and updated both nm linker and pnpify to use this package and returned back simple cli to run `yarn pnpify <program>`. I have also moved out `@yarnpkg/sdks` into a separate package. 

Open questions - what to do with pnpify-ish language inside SDKs, maybe that should be removed? What to do with `yarn sdks run` command which launches `pnpify` now, maybe it should be removed from sdk too? What to do with `--sdk` option, its a bit verbose to write `yarn sdks --sdk vscode`, maybe `--sdk` option should be removed too?

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-sdks/sources/commands/SdkCommand.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export default class SdkCommand extends Command {
     ]);
 
     if (allIntegrations.size === 0 && !onlyBase)
-      throw new UsageError(`No integrations have been provided as arguments, and no preexisting integrations could be found. Make sure to run \`yarn pnpify --sdk <integrations>\` first, or \`yarn pnpify --sdk base\` if you only need the SDK files and prefer to manage your own environment settings. Run \`yarn pnpify --sdk -h\` to see the list of supported integrations.`);
+      throw new UsageError(`No integrations have been provided as arguments, and no preexisting integrations could be found. Make sure to run \`yarn sdks --sdk <integrations>\` first, or \`yarn sdks --sdk base\` if you only need the SDK files and prefer to manage your own environment settings. Run \`yarn sdks --sdk -h\` to see the list of supported integrations.`);
```
Comment (arcanis) [inline packages/yarnpkg-sdks/sources/commands/SdkCommand.ts:112]:
```suggestion
      throw new UsageError(`No integrations have been provided as arguments, and no preexisting integrations could be found. Make sure to run \`yarn sdks <integrations>\` first, or \`yarn sdks base\` if you only need the SDK files and prefer to manage your own environment settings. Run \`yarn sdks -h\` to see the list of supported integrations.`);
```

File: packages/yarnpkg-sdks/sources/generateSdk.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export const validateIntegrations = (integrations: Set<string>) => {
   }
 
   if (unsupportedIntegrations.length > 0) {
-    throw new UsageError(`No supported integrations with the following names could be found: ${unsupportedIntegrations.join(`, `)}. Run \`yarn pnpify --sdk -h\` to see the list of supported integrations.`);
+    throw new UsageError(`No supported integrations with the following names could be found: ${unsupportedIntegrations.join(`, `)}. Run \`yarn sdks --sdk -h\` to see the list of supported integrations.`);
```
Comment (arcanis) [inline packages/yarnpkg-sdks/sources/generateSdk.ts:37]:
```suggestion
    throw new UsageError(`No supported integrations with the following names could be found: ${unsupportedIntegrations.join(`, `)}. Run \`yarn sdks -h\` to see the list of supported integrations.`);
```

File: packages/gatsby/content/getting-started/editor-sdks.md [hunk 3]
```diff
@@ -7,11 +7,11 @@ description: An overview of the editor SDKs used to bring PnP compatibility to e
 
 Smart IDEs (such as VSCode or IntelliJ) require special configuration for TypeScript to work. This page intends to be a collection of settings for each editor we've worked with - please contribute to this list!
 
-The editor SDKs and settings can be generated using the `yarn pnpify --sdk` (or `yarn dlx @yarnpkg/pnpify --sdk` if you don't need to install it locally) command. Its detailed documentation can be found on the [dedicated page](/pnpify/cli/--sdk).
+The editor SDKs and settings can be generated using the `yarn sdks` (or `yarn dlx @yarnpkg/sdks` if you don't need to install it locally) command. Its detailed documentation can be found on the [dedicated page](/sdks/cli/default).
```
Comment (arcanis) [inline packages/gatsby/content/getting-started/editor-sdks.md:10]:
```suggestion
The editor SDKs and settings can be generated using `yarn dlx @yarnpkg/sdks` (or `yarn sdks` if you added `@yarnpkg/sdks` to your dependencies). Its detailed documentation can be found on the [dedicated page](/sdks/cli/default).
```

**Review by arcanis — APPROVED** [review]
LGTM; one clarification in the documentation

**Comment by tim-stasse** [comment]
Hey @paul-soporan do you mind if I ask why the generated sdk version was changed to be fixed (i.e. `0.0.0-sdk`)? It's not really a problem, but I'm just curious what the thought process was behind it 🙂

Now the workspace's selected typescript version appears in vscode as:
image
This means now we have to go look up the typescript version in the `package.json`, where as before we could just glance at the toolbar 😅

**Comment by paul-soporan** [comment]
@tim-stasse I'll copy what @arcanis said on Discord in response to the same question:

> since the version remained out-of-date unless you re-ran the sdk (which was otherwise unnecessary), it was needlessly confusing (I myself saw a few colleagues report that they had the wrong TS version, only for the problem to just be the VSCode displayed version being incorrect)

> so in 3.0 (which is what @yarnpkg/sdks currently points for technical reasons), the version displayed is now always 0.0.0-sdk

> ideally it'd be nice to always have the correct version, but failing that a completely wrong one may be  more helpful than a likely-wrong one, depending on the perspective

---

> This means now we have to go look up the typescript version in the package.json, where as before we could just glance at the toolbar sweat_smile

This is a good point (although it can be confusing if out-of-date). I personally don't mind it either way.

@arcanis Do you have any other thoughts on this? Should we just revert it?

**Comment by Unbuttun** [comment]
One bug that I found out with the latest sdk that shows version as 0.0.0-sdk is that renaming in VsCode by pressing F2 no longer works.

But it works as long as we select the original typescript version in settings.

**Comment by TommasoAmici** [comment]
Setting the version to 0.0.0 seems to break several features of VSCode that are only available on certain Typescript versions. For example, Organize imports is not there anymore as it requires v2.8.

**Comment by paul-soporan** [comment]
Opened #3173 to revert the fixed sdk versions, we didn't originally realize that it would affect VSCode's behavior.

**Pull request merged**